### ISSUE TITLE
Issues fetching data from ESP32 into MyAmbiance App in CO2 Gadget Advanced Firmware

### ISSUE THREAD

**Issue by melkati** [melkati/CO2-Gadget#216]
Looks like there is a bug somewhere that is affecting the data fetching from ESP32 into MyAmbiance App.

After nearly 2 weeks almost 100% focused on debugging issues with the combination between ESP32 and MyAmbiance because of this bug and [another one](https://github.com/Sensirion/arduino-ble-gadget/issues/36) (I'm not sure if it's related and is temporarily solved with a workaround already) I'm completely lost at where to look.

I'm still not sure if the issue is related to the CO2 Gadget advanced firmware, the Sensirion ESP32 libs or the MyAmbiance App, I'm completely lost and clueless.

Right now, as the "oficial" Sensirion's library release is not usable for as (because of the [data gaps in MyAmbiance](https://github.com/Sensirion/arduino-ble-gadget/issues/36)) I can't be sure if the issue can be with the combination of libraries CO2 Gadget advanced firmware is using:

```
https://github.com/Sensirion/arduino-upt-core.git#b2c0e76  ; Fix for arduino-ble-gadget incompatibly with arduino-upt-core library
https://github.com/melkati/arduino-ble-gadget.git#historyInterval
```

Right now the issues with Sensirion's library and MyAmbiance are only critical things delaying the release of the new CO2 Gadget advanced firmware version.

The issue:

When fetching data from ESP32 with Sensirion's MyAmbiance the download stops at some point.

- Occasionally, data retrieval is successful, but at other times it fails.
- The issue does not consistently occur at the same point.
- This problem affects all members of the Alpha/Beta testing team.
- Sometimes, retrying results in successful data fetching, while at other times it does not.

![Screenshot_2024-05-11-07-46-05-046_com sensirion myam](https://github.com/melkati/CO2-Gadget/assets/11509521/940f3832-402a-4708-ab14-49ec849c3316)

![Screenshot_2024-05-11-07-47-03-973_com sensirion myam](https://github.com/melkati/CO2-Gadget/assets/11509521/468f3627-1033-4343-ad1e-1dbb8017cb35)

**Comment by Olaruci** [comment]
For me  a clear data & cache of the app after the update to 3.3 resolved the issue.

**Comment by melkati** [comment]
I attempted once more (having tried previously) but it continues to fail for me:

![image](https://github.com/melkati/CO2-Gadget/assets/11509521/f1fa8f89-14bb-422e-97f0-bf9a25887b31)

**Comment by Olaruci** [comment]
Yes it fails again for me also. The clear cache of app is not a fix.
And yes inconsistent samples numbers also.
![Screenshot_20240512_150553.jpg](https://github.com/melkati/CO2-Gadget/assets/47052441/28d7a17b-8bbc-4594-ac35-5599bd5f6210)

I would like to try previous version 3.2. With that it worked better for me, but I cannot find the apk online (only 2.3 that also doesn't work).
I've let also a comment in play store a comment,  hope they see it.

**Comment by Coscolin** [comment]
I have the same issue as @melkati and @Olaruci , even with very few records to download, it fails.

First attempt:
![image](https://github.com/melkati/CO2-Gadget/assets/92800807/65e4d596-e63f-4742-abf6-9a3293689861)

Second attempt:
![image](https://github.com/melkati/CO2-Gadget/assets/92800807/17980b5e-81ee-4a92-96c6-cc2a47091775)

Third atempt:
![image](https://github.com/melkati/CO2-Gadget/assets/92800807/cca6b865-b501-4f5d-80b9-5190cf321f85)

**Comment by Micci253** [comment]
it fails again.  The clear cache of Sensirion v3.3  application, there were no significant changes. Tried it on devices with Android versions 8,9, Miui 13. 
 I thought that maybe there was a connection with the amount of accumulated data - but even after a small accumulation of data, a reading error may occur. even if before this it was possible to count everything completely in several attempts.
Could the reason be due to some energy-saving features of the bluetooth?

**Comment by melkati** [comment]
I compiled the official basic example from Sensirion (Example1) and it's encountering the same issue. To ensure independence from CO2 Gadget Advanced, I installed it from Arduino IDE to isolate it from the CO2 Gadget Advanced firmware developed with VS Code and PlatformIO.

It seems like the issue lies within the library or MyAmbiance (I tested with the new MyAmbiance 3.3.1).

Steps to Reproduce:

Compile and upload the official basic example (Example1) from Sensirion.
Observe the issue persists.
Expected Behavior:
The basic example should function correctly without any data gaps or discrepancies when communicating with the Android phone via MyAmbiance.

Actual Behavior:
The same issue persists, indicating a potential issue within the library or MyAmbiance.

Additional Information:

Tested with MyAmbiance 3.3.1.
Installed from Arduino IDE for isolation from CO2 Gadget Advanced firmware.
This issue seems to occur regardless of the specific project setup, pointing towards a potential issue within the library or MyAmbiance.

![Screenshot_2024-05-18-13-14-07-263_com sensirion myam](https://github.com/melkati/CO2-Gadget/assets/11509521/26c559b9-ecc9-4102-acaa-bc3b29a1074a)

I think it's time to report this to Sensirion.
